demos/browser/app/meetingV2/meetingV2.ts

```diff
@@ -2273,15 +2284,64 @@ export class DemoMeetingApp
     this.log(`session stopped from ${JSON.stringify(sessionStatus)}`);
     this.log(`resetting stats in WebRTCStatsCollector`);
     this.statsCollector.resetStats();
+
+    const returnToStart = () => {
+      // If you want to make this a repeatable SPA, change this function to
+      // ```
+      // this.switchToFlow('flow-authenticate');
+      // ```
+      // and fix some state (e.g., video buttons).
+      //
+      // For simplicity, let's just reload the page.
+      window.location.href = window.location.pathname;
+    };
+
+    /**
+     * This is approximately the inverse of the initialization method above.
+     * This work only needs to be done if you want to continue using the page; if
+     * your app navigates away or closes the tab when done, you can let the browser
+     * clean up.
+     */
+    const onLeftMeeting = async () => {
+      // Clean up the timers for this.
+      this.audioVideo.unsubscribeFromActiveSpeakerDetector(this.activeSpeakerHandler);
```

Comment:
Is there anyway we could automatically unsubscribe once the meeting session stop? I feel It is a bit overwhelming for builders to remember to cleanup everything themselves.

Reply:
Remember that you can start, stop, start an AV session.

The active speaker detector doesn't know whether the session is started or stopped, and the detector is created and passed as an argument to the AV facade.

At one point I had `DefaultAudioVideoController.stop` disposing of the active speaker detector, but that would require that the caller re-subscribe before the next `start`, and that's simply not our API contract.

Further, unsubscribing here helps to avoid leaking `this.activeSpeakerHandler`, which is retained in the detector.

Happy to have someone who knows more about the guts here suggest a way to get this to avoid leaking — maybe there's a realtime observer notification that would allow the detector to discard the timers and recreate them?

Reply:
For simplicity, can we just have a dispose/clean up function for AV to unsubscribe everything ?

Reply:
I don't think that's as easy as we might want.

We couldn't just unsubscribe, because we need to know which handlers to unsubscribe, and this is what the code looks like when it does!

We could add an `unsubscribeAll` method to a bunch of places, but that makes our interface bigger and isn't feasible for classes that don't keep a reference to their callbacks — `activeSpeakerHandler` does, but not all do.

Implementing `destroy` on the facade is trickier than you'd think. Because of the facade pattern, we would need `isDestroyable` checks and `destroy` calls on every component member of `DefaultAudioVideoFacade`. Each of the default classes would need to implement `Destroyable` and recurse down.

There are also confusing object references in the demo — e.g., `this.meetingSession` holds a reference to the device controller, which cleaned up by a destroyer — and so we would save ~5 lines of code here, but still need to do some cleanup as you can see below.

I am not opposed to doing this work, but I think it's a good intermediate step to make manual cleanup like this function correctly, and then consider whether making `DefaultAudioVideoFacade implements Destroyable` is a worthwhile step.